src/main.rs

```diff
@@ -197,28 +198,29 @@ async fn root_get() -> Html<&'static str> {
 
 // TODO: create tests for endpoints: #38
 async fn root_post(
-    claims: auth::Claims,
+    claims: Result<Claims, ClaimsError>,
     mut multipart: Multipart,
     command: String,
     timeout: u64,
     jobs: usize,
-) -> Result<impl IntoResponse, StatusCode> {
+) -> impl IntoResponse {
+    let claims = claims.map_err(|e| e.redirect_response().into_response())?;
     let user = claims.subject().to_string();
 
     // Detect too many jobs early.
     {
         let lock = OUT.read().await;
 
         if lock.len() >= jobs {
-            return Err(StatusCode::TOO_MANY_REQUESTS);
+            return Err(redirect::too_many_workloads().into_response());
```

Comment:
Are all the explicit `into_response()` calls necessary throughout this function, since this return type is `impl IntoResponse`?

Reply:
Yeah because StatusCode::BAD_REQUEST and Redirect are different types and will give you

```
expected struct `axum::response::Redirect`, found struct `axum::http::StatusCode`
```

a bunch of times

Reply:
Yeah because StatusCode::BAD_REQUEST and Redirect are different types and will give you

```
expected struct `axum::response::Redirect`, found struct `axum::http::StatusCode`
```

a bunch of times